src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -18,8 +18,10 @@ public abstract class FileCleanupTestBase : IDisposable
         protected static bool IsProcessElevated => s_isElevated.Value;
 
         /// <summary>Initialize the test class base.  This creates the associated test directory.</summary>
-        protected FileCleanupTestBase()
+        protected FileCleanupTestBase(string cleanupRootDirectory = null)
         {
+            cleanupRootDirectory ??= Path.GetTempPath();
+
             // Use a unique test directory per test class.  The test directory lives in the user's temp directory,
```

Comment:
Nit, it no longer lives in temp directory necessarily.

Or you could just rename cleanupRootDirectory->tempDirectory